libcxx/include/condition_variable

```diff
@@ -257,23 +258,38 @@ condition_variable_any::wait_for(_Lock& __lock, const chrono::duration<_Rep, _Pe
 
 template <class _Lock, class _Predicate>
 bool condition_variable_any::wait(_Lock& __lock, stop_token __stoken, _Predicate __pred) {
-  while (!__stoken.stop_requested()) {
+  shared_ptr<mutex> __mut = __mut_;
```

Comment:
I am not sure what we really want to take the copy of ownership of the mutex in these overloads. By taking the copy, it sometimes works and sometimes it does not . (If notified by the request_stop followed by the destructor call, it is UB on accessing the internal cv)

if we were allowed to break the ABI, we would bundle the cv and mutex  into the shared_ptr. but we are not.  I am not sure if we just not taking the copy at all and claim we don’t support destroy-while-waiting, rather than here sometimes it works and sometimes it doesn’t.

Reply:
The only reason I did this was to stay consistent with the other condition_variable_any::wait calls that don't accept a stop_token, because they are currently copying the internal mutex pointer. If you don't think it's desirable/useful I can update the PR to remove it. 

Although I'm not sure if you'd rather update #77127 to include the internal lock lifetime changes instead, and drop this PR?

Reply:
I don't mind to be honest. It is bit unlucky that we both worked on this on the same day and ended up with two patches. I updated mine with your suggestion and added test that you discovered (request_stop in between check and wait).
if we are aiming to land this patch, it would be good to have some tests (maybe you can take mine as starting point). Let me know your thoughts and thank you for contribution